rustls/src/crypto/mod.rs

```diff
@@ -45,19 +45,137 @@ pub use crate::rand::GetRandomFailed;
 
 pub use crate::msgs::handshake::KeyExchangeAlgorithm;
 
-/// Pluggable crypto galore.
+/// Controls core cryptography used by rustls.
+///
+/// This crate comes with two built-in options, provided as
+/// `&dyn CryptoProvider` values:
+///
+/// - [`crate::crypto::ring::RING`]: (behind the `ring` crate feature, which
+///   is enabled by default).  This provider uses the [*ring*](https://github.com/briansmith/ring)
+///   crate.
+/// - [`crate::crypto::aws_lc_rs::AWS_LC_RS`]: (behind the `aws_lc_rs` feature,
+///   which is optional).  This provider uses the [aws-lc-rs](https://github.com/aws/aws-lc-rs)
+///   crate.
+///
+/// # Using a specific `CryptoProvider`
+/// Supply the provider when constructing your [`crate::ClientConfig`] or [`crate::ServerConfig`]:
+///
+/// - [`crate::ClientConfig::builder_with_provider()`]
+/// - [`crate::ServerConfig::builder_with_provider()`]
+///
+/// When creating and configuring a webpki-backed client or server certificate verifier, the provider
+/// is also needed to start:
+///
+/// - [`crate::client::WebPkiServerVerifier::builder_with_provider()`]
+/// - [`crate::server::WebPkiClientVerifier::builder_with_provider()`]
+///
+/// # Making a custom `CryptoProvider`
+/// Naturally start with a type that `impl rustls::crypto::CryptoProvider`.
```

Comment:
I think this reads a little awkwardly and the `impl` bit of syntax is probably well understood:
```suggestion
/// Naturally start with a type that implements `rustls::crypto::CryptoProvider`.
```